fixes a bug in libtool versions < 2.4/1.3293
the sysroot code returns =/lib/libfoo.so instead of $lt_sysroot/lib/libfoo.so

--- libXScrnSaver-1.2.2/libtool
+++ ../libsm/libSM-1.2.2/libtool
@@ -9684,7 +9701,8 @@
 	      *.la)
 		func_basename "$deplib"
 		name="$func_basename_result"
-		eval libdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
+		func_resolve_sysroot "$deplib"
+		eval libdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $func_resolve_sysroot_result`
 		test -z "$libdir" && \
 		  func_fatal_error "\`$deplib' is not a valid libtool archive"
 		newdependency_libs+=" ${lt_sysroot:+=}$libdir/$name"
